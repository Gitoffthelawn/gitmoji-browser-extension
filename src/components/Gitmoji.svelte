<script>
  export let code;
  export let emoji;
  export let description;
  export let color;

  const copyToClipboard = (value) => {
    navigator.clipboard.writeText(value).then(function() {
      console.log('Async: Copying to clipboard was successful!');
    }, function(err) {
      console.error('Async: Could not copy text: ', err);
    });
  }
</script>

<li class="gitmoji" style="--color:{color}">
  <button 
    class="emoji"
    on:click={() => { copyToClipboard(emoji) }}
  >
    <span>{emoji}</span>
  </button>
  <button 
    class="code"
    on:click={() => { copyToClipboard(code) }}>
    <span>{code}</span>
  </button>
</li>

<style>
button {
	background: none;
	color: inherit;
	border: none;
	padding: 0;
	font: inherit;
	cursor: pointer;
	outline: inherit;
}

.gitmoji {
  display: flex;
  margin: 0.5em;
  box-shadow: 0 1px 2px 0 rgba(168,182,191,.6);
  border-radius: 4px;
}

.emoji {
  font-size: 2em;
  padding: 0.15em 0.4em calc(0.4em - 5px) 0.4em;
  background: var(--color);
  border-radius: 4px 0 0 4px;
  border-bottom: solid 3px transparent;
}

.emoji span {
  display: block;
}

.emoji:hover span,
.emoji:focus span {
  animation-name: bounce;
  animation-duration: 0.5s;
}

.code {
  width: 100%;
  font-size: 1.5em;
  padding-left: 0.5em;
  font-weight: 450;
  text-align: left;
}

.code span {
  border-radius: 4px;
  transition-duration: .3s;
  border-bottom: 2px solid transparent;
  background-color: rgba(var(--color), 0.3);
  box-shadow: 0 4px rgba(var(--color), 0.6);
}

.code:hover,
.code:focus {
  transition-duration: .3s;
  /* background-color: var(--color);
  border-left: 1px solid black; */
}

/*
  This code has been obtained from:
  https://github.com/daneden/animate.css/blob/master/source/attention_seekers/bounce.css
*/
@keyframes bounce {
  from,
  20%,
  53%,
  80%,
  to {
    animation-timing-function: cubic-bezier(0.215, 0.61, 0.355, 1);
    transform: translate3d(0, 0, 0);
  }

  40%,
  43% {
    animation-timing-function: cubic-bezier(0.755, 0.05, 0.855, 0.06);
    transform: translate3d(0, -.2em, 0);
  }

  70% {
    animation-timing-function: cubic-bezier(0.755, 0.05, 0.855, 0.06);
    transform: translate3d(0, -.1em, 0);
  }

  90% {
    transform: translate3d(0, -.05em, 0);
  }
}

</style>