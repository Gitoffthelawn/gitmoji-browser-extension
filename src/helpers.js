export const onGetTabInnerHTML = (callback) => {
  if (chrome.tabs !== undefined) {
    chrome.tabs.executeScript(null, { code: 'document.body.innerText' },
      (results) => { callback(results[0]) })
  }
}
