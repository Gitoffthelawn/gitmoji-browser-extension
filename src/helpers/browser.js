/* eslint-disable no-undef */
export const onGetTabInnerHTML = (callback) => {
  if (typeof chrome !== 'undefined' && chrome.tabs !== undefined) {
    chrome.tabs.executeScript(null, { code: 'document.body.innerText' },
      (results) => { callback(results[0]) })
  }
}

export const isInExtension = () => {
  return typeof chrome !== 'undefined' && chrome.tabs !== undefined
}
