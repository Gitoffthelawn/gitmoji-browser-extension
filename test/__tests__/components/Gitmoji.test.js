import { render, fireEvent, screen } from '@testing-library/svelte'
import Gitmoji from '../../../src/components/Gitmoji.svelte'
import * as clipboard from '../../../src/helpers/clipboard'

// ðŸ¤¡ Mock clipboard object
jest.mock('../../../src/helpers/clipboard')

describe('<Gitmoji />', () => {
  let wrapper
  let onSaveSpy = jest.fn()
  let gitmojiProps = {
    code: ':art:',
    emoji: 'ðŸŽ¨',
    color: '#ff7281',
    description: 'Improve structure / format of the code.',
    onSave: onSaveSpy
  }

  beforeEach(() => {
    wrapper = render(Gitmoji, gitmojiProps)
  })

  describe('snapshot', () => {
    it('should match snapshot', () => {
      const { container } = wrapper

      expect(container).toMatchSnapshot()
    })

    it('should match snapshot if is present', () => {
      const { container } = render(Gitmoji, { ...gitmojiProps, present: true })

      expect(container).toMatchSnapshot()
    })
  })

  describe('save in clipboard', () => {
    it('should save emoji in clipboard', async () => {
      const { container } = wrapper
      const saveEmojiBtn = screen.getByRole('button', { name: gitmojiProps.emoji })

      await fireEvent.click(saveEmojiBtn)

      expect(clipboard.setClipboard).toHaveBeenCalledWith(gitmojiProps.emoji)
      expect(onSaveSpy).toHaveBeenCalled()
    })

    it('should save shortcode in clipboard', async () => {
      const { container } = wrapper
      const saveShortcodeBtn = screen.getByRole('button', { name: gitmojiProps.code, hidden: true })

      await fireEvent.click(saveShortcodeBtn)

      expect(clipboard.setClipboard).toHaveBeenCalledWith(gitmojiProps.code)
      expect(onSaveSpy).toHaveBeenCalled()
    })
  })
})
