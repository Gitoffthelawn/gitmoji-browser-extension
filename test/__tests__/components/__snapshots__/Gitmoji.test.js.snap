// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gitmoji /> snapshot should match snapshot 1`] = `
<body>
  <div>
    <li
      class="gitmoji"
      style="--color: #ff7281;"
    >
      <button
        class="emoji"
      >
        <span>
          ðŸŽ¨
        </span>
      </button>
       
      <button
        class="code"
      >
        <span>
          :art:
        </span>
      </button>
       
      <div
        class="indicator-group"
      />
    </li>
  </div>
  <div>
    <li
      class="gitmoji"
      style="--color: #ff7281;"
    >
      <button
        class="emoji"
      >
        <span>
          ðŸŽ¨
        </span>
      </button>
       
      <button
        class="code"
      >
        <span>
          :art:
        </span>
      </button>
       
      <div
        class="indicator-group"
      />
    </li>
  </div>
</body>
`;

exports[`<Gitmoji /> snapshot should match snapshot if is present 1`] = `
<body>
  <div>
    <li
      class="gitmoji"
      style="--color: #ff7281;"
    >
      <button
        class="emoji"
      >
        <span>
          ðŸŽ¨
        </span>
      </button>
       
      <button
        class="code"
      >
        <span>
          :art:
        </span>
      </button>
       
      <div
        class="indicator-group"
      />
    </li>
  </div>
  <div>
    <li
      class="gitmoji"
      style="--color: #ff7281;"
    >
      <button
        class="emoji"
      >
        <span>
          ðŸŽ¨
        </span>
      </button>
       
      <button
        class="code"
      >
        <span>
          :art:
        </span>
      </button>
       
      <div
        class="indicator-group"
      >
        <div
          title="this gitmoji appears in the page"
        >
          <img
            alt="icon: eye"
            class="icon eye eye"
          />
        </div>
      </div>
    </li>
  </div>
</body>
`;
